homeassistant/components/switch/raspyrfm.py

```diff
@@ -0,0 +1,157 @@
+"""
+Support for switch devices that can be controlled using the RaspyRFM rc module.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch/raspyrfm/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, CONF_PLATFORM, CONF_PORT, CONF_SWITCHES,
+    DEVICE_DEFAULT_NAME)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['raspyrfm-client==1.2.8']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_GATEWAY_MANUFACTURER = 'gateway_manufacturer'
+CONF_GATEWAY_MODEL = 'gateway_model'
+CONF_CONTROLUNIT_MANUFACTURER = 'controlunit_manufacturer'
+CONF_CONTROLUNIT_MODEL = 'controlunit_model'
+CONF_CHANNEL_CONFIG = 'channel_config'
+DEFAULT_HOST = '127.0.0.1'
+
+# define configuration parameters
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): 'raspyrfm',
+    vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=None):
+        vol.All(vol.Coerce(int), vol.Range(min=1, max=65535)),
+    vol.Optional(CONF_GATEWAY_MANUFACTURER): cv.string,
+    vol.Optional(CONF_GATEWAY_MODEL): cv.string,
+    vol.Required(CONF_SWITCHES, default=[]): vol.Schema([{
+        vol.Optional(CONF_NAME, default=DEVICE_DEFAULT_NAME): cv.string,
+        vol.Required(CONF_CONTROLUNIT_MANUFACTURER): cv.string,
+        vol.Required(CONF_CONTROLUNIT_MODEL): cv.string,
+        vol.Required(CONF_CHANNEL_CONFIG): cv.match_all,
+    }])
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the RaspyRFM switch."""
+    _LOGGER.info("initializing RaspyRFM Switch")
+
+    from raspyrfm_client import RaspyRFMClient
+    from raspyrfm_client.device_implementations.controlunit. \
+        controlunit_constants import ControlUnitModel
+    from raspyrfm_client.device_implementations.gateway.manufacturer. \
+        gateway_constants import GatewayModel
+    from raspyrfm_client.device_implementations.manufacturer_constants \
+        import Manufacturer
+
+    # read configuration
+    gateway_manufacturer = config.get(CONF_GATEWAY_MANUFACTURER,
+                                      Manufacturer.SEEGEL_SYSTEME.value)
+    gateway_model = config.get(CONF_GATEWAY_MODEL, GatewayModel.RASPYRFM.value)
+    host = config.get(CONF_HOST, DEFAULT_HOST)
```

Comment:
Default host should be set in the config schema.

Reply:
Yes